### DISCUSSION TITLE
How to make a filament Builder fields translatable using the spatie plugin ?

### DISCUSSION THREAD

**Discussion by alibbali** [filamentphp/filament#12313] [Help]
### Package

Form builder

### Package Version

v3.x

### How can we help you?

Hello everyone,
I'm using filament v3
I'm using the field builder (https://filamentphp.com/docs/3.x/forms/fields/builder)

I have the spatie translatable plugin in my project to translate my data, but all of the data inside the builder block arent translated. 

For example, my slug field data is translated ({"fr":"a-test"})
While my builder field block isnt ([{"type":"heading","data":{"content":"A test","level":"h1"}}])

Is there any solution to make this Builder fields translatable ? 

Thank you !

**Comment by leandrocfe** [comment]
Did you add the builder attribute to the `$translatable` variable in your Model?

**Reply by alibbali** [reply]
Yes, my $attribute "content" is in the array :/
Builder fields creates a specific structure and saves it in this json field "content", but without the translated value

**Reply by leandrocfe** [reply]
Hum.. my understanding is that in this case, it’s intended to work with relationships. It appears that Spatie doesn’t provide direct support for JSON fields:

https://github.com/spatie/laravel-translatable/discussions/277

**Reply by alibbali** [reply]
Ok thank you Leandro 👍

**Comment by lukacavic** [comment]
Hello, I have problem where other languages are json object, and current active language is json array. Also, when editing active language, data is stored ok, but other locales are not, like component uuid is generated as json object, and that object has builder data. 
"hr" element is object and "en" is array. "en" is activeLanguage and "hr" is otherLocale

```json
{
  "hr": {
    "9de5a1af-6f2a-4291-b433-c1939ecdb354": {
      "type": "heading-element",
      "data": {
        "content": "dwadwwokšpk",
        "level": "h1"
      }
    }
  },
  "en": [
    {
      "type": "heading-element",
      "data": {
        "content": "iuhihuh",
        "level": "h3"
      }
    }
  ]
}
```
